src/App.tsx

```diff
@@ -1,26 +1,257 @@
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTempTodo(null);
+        setTodoTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTempTodo(null);
+        setTodoTitle(newTempTodo.title);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+
+    return service.deleteTodos(todoId).catch(() => {
+      setTodos(todos);
+      setErrorMessage('Unable to delete a todo');
+    });
+  };
+
+  const deleteCompleted = (completed: Todo[]) => {
+    for (const completedTodo of completed) {
+      if (completedTodo.completed) {
+        deleteTodo(completedTodo.id);
+      }
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    addTodo({ title: todoTitle });
+    setTodoTitle('');
+  };
+
+  const handleButtonClose = () => {
+    setErrorMessage('');
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed */}
+          {todos.length !== 0 && (
+            <button
+              type="button"
+              className={classNames('todoapp__toggle-all', {
+                active: completedTodos.length === todos.length,
+              })}
+              data-cy="ToggleAllButton"
+              onClick={() => {
+                if (completedTodos.length === todos.length) {
+                  deleteCompleted(todos);
+                }
+              }}
+            />
+          )}
+
+          {/* Add a todo on form submit */}
+          <form onSubmit={handleSubmit}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={todoTitle}
+              onChange={handleTitleChange}
+              autoFocus
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main" data-cy="TodoList">
+          {filteredTodos.map(todo => (
+            <TodoItem key={todo.id} todo={todo} onDelete={deleteTodo} />
+          ))}
+          {tempTodo && <TodoItem todo={tempTodo} />}
+        </section>
+
+        {/* Hide the footer if there are no todos */}
+        {todos.length !== 0 && (
```

Comment:
move footer into a separate component to separate logic and layout a bit into a separate components